server/persistence/mutate/access.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2024 The Workspaces Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutate
+
+import (
+	"context"
+
+	restworkspacesv1alpha1 "github.com/konflux-workspaces/workspaces/server/api/v1alpha1"
+	"github.com/konflux-workspaces/workspaces/server/persistence/clientinterface"
+)
+
+// Applies the has-direct-access label if the user "accessor" has access via
+// public viewer or via a direct binding to the workspace
+func ApplyHasDirectAccessLabel(
+	ctx context.Context,
+	client clientinterface.InternalWorkspacesReadClient,
+	workspace *restworkspacesv1alpha1.Workspace,
```

Comment:
To avoid the `clientinterface` dependency we could
```suggestion
	restworkspacesv1alpha1 "github.com/konflux-workspaces/workspaces/server/api/v1alpha1"
)

type DirectAccessChecker interface {
    UserHasDirectAccess(context.Context, string, string) (bool, error)
}

// Applies the has-direct-access label if the user "accessor" has access via
// public viewer or via a direct binding to the workspace
func ApplyHasDirectAccessLabel(
	ctx context.Context,
	client DirectAccessChecker,
	workspace *restworkspacesv1alpha1.Workspace,
```

Reply:
I don't know if it makes sense to have this interface live underneath `mutate`, since ideally I'd have that package contain implementations of functions that do post-processing on workspaces on their way back from the api server.  Semantically, the access checker was something that already lived as a part of the reading client, I'm just looking to expose it with an interface (which IMO lives alongside the definitions of the other reading interfaces).